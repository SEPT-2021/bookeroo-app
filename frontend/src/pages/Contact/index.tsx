import React, { useState } from "react";
import {
  Button,
  Card,
  CardContent,
  createStyles,
  Grid,
  TextField,
  Typography,
  withStyles,
  WithStyles,
} from "@material-ui/core";

function Contact({ classes }: ContactProps) {
  const [isSubmitted, setSubmitted] = useState(false);
  return (
    <div className={classes.root}>
      <Grid>
        <Card style={{ maxWidth: 450, padding: "20px 5px", margin: "0 auto" }}>
          <CardContent>
            <Typography gutterBottom variant="h5">
              Contact Us
            </Typography>
            <Typography
              variant="body2"
              color="textSecondary"
              component="p"
              gutterBottom
            >
              Fill up the form and our team will get back to you within 24
              hours.
            </Typography>
            <form>
              <Grid container spacing={1}>
                <Grid xs={12} sm={6} item>
                  <TextField
                    placeholder="Enter first name"
                    label="First Name"
                    variant="outlined"
                    fullWidth
                    required
                  />
                </Grid>
                <Grid xs={12} sm={6} item>
                  <TextField
                    placeholder="Enter last name"
                    label="Last Name"
                    variant="outlined"
                    fullWidth
                    required
                  />
                </Grid>
                <Grid item xs={12}>
                  <TextField
                    type="email"
                    placeholder="Enter email"
                    label="Email"
                    variant="outlined"
                    fullWidth
                    required
                  />
                </Grid>
                <Grid item xs={12}>
                  <TextField
                    label="Message"
                    multiline
                    rows={4}
                    placeholder="Type your message here"
                    variant="outlined"
                    fullWidth
                    required
                  />
                </Grid>
                <Grid item xs={12}>
                  <Button
                    onClick={() => {
                      setSubmitted(true);
                      setTimeout(
                        () => window.open("mailto:hello@bookeroo.com.au"),
                        1000
                      );
                    }}
                    disabled={isSubmitted}
                    type="submit"
                    variant="contained"
                    color="primary"
                    fullWidth
                  >
                    {isSubmitted
                      ? "Done! We'll get back to you shortly"
                      : "Submit"}
                  </Button>
                </Grid>
              </Grid>
            </form>
          </CardContent>
        </Card>
      </Grid>
    </div>
  );
}

const styles = () => {
  return createStyles({
    root: {
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      marginTop: "120px",
      height: "100vh",
    },
  });
};

type ContactProps = WithStyles<typeof styles>;

export default withStyles(styles)(Contact);
