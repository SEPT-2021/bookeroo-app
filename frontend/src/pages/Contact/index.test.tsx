import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import { QueryClient, QueryClientProvider } from "react-query";
import Contact from "./index";

afterEach(cleanup);
describe("Rendering page Contact", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <QueryClientProvider client={new QueryClient()}>
        <Contact />
      </QueryClientProvider>,
      div
    );
    expect(div).toMatchSnapshot();
  });
});

describe("Contact matches snapshot", () => {
  test("render DeleteBook", () => {
    const { container } = render(
      <QueryClientProvider client={new QueryClient()}>
        <Contact />
      </QueryClientProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
