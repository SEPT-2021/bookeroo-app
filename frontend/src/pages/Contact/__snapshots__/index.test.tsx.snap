// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact matches snapshot render DeleteBook 1`] = `
<div>
  <div
    class="Contact-root-1"
  >
    <div
      class="MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
        style="max-width: 450px; padding: 20px 5px; margin: 0px auto;"
      >
        <div
          class="MuiCardContent-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
          >
            Contact Us
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
          >
            Fill up the form and our team will get back to you within 24 hours.
          </p>
          <form>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                    data-shrink="false"
                  >
                    First Name
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      placeholder="Enter first name"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-4"
                      >
                        <span>
                          First Name
                           *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                    data-shrink="false"
                  >
                    Last Name
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      placeholder="Enter last name"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-4"
                      >
                        <span>
                          Last Name
                           *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                    data-shrink="false"
                  >
                    Email
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      placeholder="Enter email"
                      required=""
                      type="email"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-4"
                      >
                        <span>
                          Email
                           *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                    data-shrink="false"
                  >
                    Message
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
                  >
                    <textarea
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                      placeholder="Type your message here"
                      required=""
                      rows="4"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-4"
                      >
                        <span>
                          Message
                           *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
                  tabindex="0"
                  type="submit"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Submit
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Rendering page Contact renders without crashing 1`] = `
<div>
  <div
    class="Contact-root-1"
  >
    <div
      class="MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
        style="max-width: 450px; padding: 20px 5px; margin: 0px auto;"
      >
        <div
          class="MuiCardContent-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
          >
            Contact Us
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
          >
            Fill up the form and our team will get back to you within 24 hours.
          </p>
          <form>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                    data-shrink="false"
                  >
                    First Name
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      placeholder="Enter first name"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-4"
                      >
                        <span>
                          First Name
                           *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                    data-shrink="false"
                  >
                    Last Name
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      placeholder="Enter last name"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-4"
                      >
                        <span>
                          Last Name
                           *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                    data-shrink="false"
                  >
                    Email
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      placeholder="Enter email"
                      required=""
                      type="email"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-4"
                      >
                        <span>
                          Email
                           *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                    data-shrink="false"
                  >
                    Message
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
                  >
                    <textarea
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                      placeholder="Type your message here"
                      required=""
                      rows="4"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-4"
                      >
                        <span>
                          Message
                           *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
                  tabindex="0"
                  type="submit"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Submit
                  </span>
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
